docs/reasonreactcompat.md

```diff
@@ -0,0 +1,222 @@
+---
+title: ReasonReactCompat: migrating to 0.7.0 and JSX v3
+---
+
+In version 0.7.0, two large changes were introduced:
+
+- Support for React hooks
+- A new JSX version (v3)
+
+You can read more in the release [blog post](https://reasonml.github.io/reason-react/blog/2019/04/10/react-hooks).
+
+To enable the progressive migrations of existing ReasonReact applications to the new API and JSX version, a new module called `ReasonReactCompat` was added to ReasonReact.
+
+## React and ReasonReact modules
+
+`ReasonReactCompat` is a bridge between two modules:
+
+- `ReasonReact`: contains the now frozen API that was used to implement components in v0.6.0 and earlier using JSX v2.
+- `React`: the module that contains the hooks-compatible API, that is available since v0.7.0 and uses JSX v3.
+
+JSX versions 2 and 3 are coupled with the corresponding modules `ReasonReact` and `React`. In other words: a component created using `ReasonReact` module will require JSX v2, and a component built using `React` module will require JSX v3.
+
+For clarity and brevity, in the rest of the section we will refer to each group as follows:
+- `ReasonReact` components (< 0.0.7) that use JSX v2, as **v2 components**
+- `React` components (>= 0.0.7) that use JSX v3, as **v3 components**.
+
+## `wrapReactForReasonReact`: Wrapping a v3 component to be used from a v2 component
+
+In most cases, we will want to migrate our application to use v3 components (see the section about migration strategies [below](https://reasonml.github.io/reason-react/docs/en/reasonreactcompat.html#migrating-an-application-to-v070-and-jsx-v3)).
+
+For example, a v2 component `Banner` might need to use a v3 component `Image`. `Image` can be made compatible by leveraging `ReasonReactCompat.wrapReactForReasonReact`:
+
+```reason
+/* In Image.re */
+[@bs.config {jsx: 3}];
+
+[@react.component]
+let make = (~src) => <img src />;
+
+module Jsx2 = {
+  let component = ReasonReact.statelessComponent("Image");
+  /* `children` is not labelled, as it is a regular parameter in version 2 of JSX */
+  let make = (~src, children) =>
+    ReasonReactCompat.wrapReactForReasonReact(
+      make,
+      makeProps(~src, ()),
+      children,
```

Comment:
For this example, wouldn't children be underscored and an empty array used here?

Reply:
I like your reasoning!  I'd also like to know @rickyvetter's thoughts.